src/string.cr

```diff
@@ -1420,8 +1420,8 @@ class String
   # is not included in the *hash*, nothing is replaced.
   #
   # ```
-  # "hello".sub(/(he|l|o)/, {"he": "ha", "l": "la"}) # => "hallo"
-  # "hello".sub(/(he|l|o)/, {"l": "la"})             # => "hello"
+  # "hello".sub(/(he|l|o)/, {"he" => "ha", "l" => "la"}) # => "hallo"
```

Comment:
@Sija 
Because it seems `NamedTuple` is not a `Hash` recently.
So then, unexpected `sub(Regex, Object)` is called.
```crystal
{"he": "ha"}.is_a?(Hash) # => false
"hello".sub(/(he|l|o)/, {"he": "ha", "l": "la"}) # => "{he: \"ha\", l: \"la\"}llo"
```
Of cource, we need to declare `String#sub(Regex, NamedTuple)` correctly later.